openmls/src/extensions/mod.rs

```diff
@@ -166,6 +166,200 @@ pub enum Extension {
     ExternalSenders(ExternalSendersExtension),
 }
 
+/// A list of extensions with unique extension types.
+#[derive(
+    Debug, Clone, PartialEq, Eq, Serialize, Deserialize, TlsSize, TlsSerialize, TlsDeserialize,
+)]
+pub struct Extensions {
+    inner: Vec<Extension>,
+}
+
+impl Extensions {
+    /// Create an extension list that is empty.
+    pub fn empty() -> Self {
+        Self { inner: Vec::new() }
+    }
+
+    /// Create an extension list that contains a single extension.
+    pub fn single(extension: Extension) -> Self {
+        Self {
+            inner: vec![extension],
+        }
+    }
+
+    /// Create an extension list that contains multiple extensions.
+    ///
+    /// This function will fail when the list of extensions contains duplicate extension types.
+    pub fn multi(extensions: Vec<Extension>) -> Result<Self, InvalidExtensionError> {
+        extensions.try_into()
+    }
+
+    // ---------------------------------------------------------------------------------------------
+
+    /// Add an extension to the extension list.
+    ///
+    /// Returns an error when there already is an extension with the same extension type.
+    pub fn add(&mut self, extension: Extension) -> Result<(), InvalidExtensionError> {
+        if !self.contains(extension.extension_type()) {
```

Comment:
Switch the cases around to have an early return.

Reply:
Can you say more? I assume that a successful add will be the happy path?

Reply:
Thanks, havn't thought about readability here. Makes sense :+1: